python/lsst/fgcmcal/fgcmCalibrateTract.py

```diff
@@ -20,462 +20,43 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
-"""Run fgcmcal on a single tract.
-
+"""Class for running fgcmcal on a single tract using src tables.
 """
 
-import sys
-import traceback
-
-import numpy as np
-
 import lsst.pex.config as pexConfig
 import lsst.pipe.base as pipeBase
 from lsst.jointcal.dataIds import PerTractCcdDataIdContainer
 
-from .fgcmBuildStars import FgcmBuildStarsTask
-from .fgcmFitCycle import FgcmFitCycleConfig
-from .fgcmOutputProducts import FgcmOutputProductsTask
-from .utilities import makeConfigDict, translateFgcmLut, translateVisitCatalog
-from .utilities import computeCcdOffsets, computeApertureRadius, extractReferenceMags
-from .utilities import makeZptSchema, makeZptCat
-from .utilities import makeAtmSchema, makeAtmCat
-from .utilities import makeStdSchema, makeStdCat
-
-import fgcm
-
-__all__ = ['FgcmCalibrateTractConfig', 'FgcmCalibrateTractTask', 'FgcmCalibrateTractRunner']
+from .fgcmCalibrateTractBase import (FgcmCalibrateTractConfigBase, FgcmCalibrateTractRunner,
+                                     FgcmCalibrateTractBaseTask)
 
+__all__ = ['FgcmCalibrateTractConfig', 'FgcmCalibrateTractTask']
 
-class FgcmCalibrateTractConfig(pexConfig.Config):
-    """Config for FgcmCalibrateTract"""
-
-    fgcmBuildStars = pexConfig.ConfigurableField(
-        target=FgcmBuildStarsTask,
-        doc="Task to load and match stars for fgcm",
-    )
-    fgcmFitCycle = pexConfig.ConfigField(
-        dtype=FgcmFitCycleConfig,
-        doc="Config to run a single fgcm fit cycle",
-    )
-    fgcmOutputProducts = pexConfig.ConfigurableField(
-        target=FgcmOutputProductsTask,
-        doc="Task to output fgcm products",
-    )
-    convergenceTolerance = pexConfig.Field(
-        doc="Tolerance on repeatability convergence (per band)",
-        dtype=float,
-        default=0.005,
-    )
-    maxFitCycles = pexConfig.Field(
-        doc="Maximum number of fit cycles",
-        dtype=int,
-        default=5,
-    )
-    doDebuggingPlots = pexConfig.Field(
-        doc="Make plots for debugging purposes?",
-        dtype=bool,
-        default=False,
-    )
 
+class FgcmCalibrateTractConfig(FgcmCalibrateTractConfigBase):
+    """Config for FgcmCalibrateTract task"""
     def setDefaults(self):
         pexConfig.Config.setDefaults(self)
+        super().setDefaults()
```

Comment:
Whoops, didn't see this one before: you only need to call `super`, not the other `setDefaults` just above. `super` takes care of everything in the MRO.